ci/build.sh

```diff
@@ -16,9 +16,9 @@ set -ex
 
 export UBSAN_OPTIONS=print_stacktrace=1
 
-cd "$BOOST_ROOT"
-
+pushd "$BOOST_ROOT"
```

Comment:
I'd undo this. This script is meant to be run, not sourced, so a change to the PWD is ok.